lib/srv/sess.go

```diff
@@ -1656,22 +1635,26 @@ func (p *party) closeUnderSessionLock() {
 	})
 }
 
-func (s *session) trackerGet() (types.SessionTracker, error) {
-	// get the session from the registry
-	sess, err := s.registry.SessionTrackerService.GetSessionTracker(s.serverCtx, s.id.String())
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
+// trackSession creates a new session tracker for the ssh session.
+// While ctx is open, the session tracker's expiration will be extended
+// on an interval. Once the ctx is closed, the session tracker's state
+// will be updated to terminated.
+func (s *session) trackSession(teleportUser string, policySet []*types.SessionTrackerPolicySet) error {
```

Comment:
There's a reasonable amount of repetition throughout the PR - trackSession methods and `sessionTrackerServiceWithCond` come to mind as non-trivial amounts of similar-looking code.

Do you think it's possible to reuse a bit more, in a way that the end result would be equal (or better!) than the present?

Reply:
That was my initial intention with the helpers in `services`, but it was a swing and a miss on the first attempt. I moved all the helper methods to the new `srv.SessionTracker`, which turned out much better.